### ISSUE TITLE
Add documentation comments to core library cadl types

### ISSUE THREAD

**Issue by markcowl** [microsoft/cadl#1226]
Dependent on #719 
- [x] Add developer comments to core library public cadl types
- [ ] Add developer comments to core library decorators

**Comment by allenjzhang** [comment]
On a related note, it would be nice to have comments on all TS artifacts for core packages.

**Comment by markcowl** [comment]
est: 3

**Comment by markcowl** [comment]
@allenjzhang see #1227
